src/Propel/Runtime/ActiveQuery/QueryExecutor/CountQueryExecutor.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+/**
+ * MIT License. This file is part of the Propel package.
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Propel\Runtime\ActiveQuery\QueryExecutor;
+
+use Propel\Runtime\ActiveQuery\Criteria;
+use Propel\Runtime\ActiveQuery\SqlBuilder\CountQuerySqlBuilder;
+use Propel\Runtime\Connection\ConnectionInterface;
+use Propel\Runtime\DataFetcher\DataFetcherInterface;
+
+/**
+ * This class produces the base object class (e.g. BaseMyTable) which contains
+ * all the custom-built accessor and setter methods.
+ */
+class CountQueryExecutor extends AbstractQueryExecutor
+{
+    /**
+     * @param \Propel\Runtime\ActiveQuery\Criteria $criteria
+     * @param \Propel\Runtime\Connection\ConnectionInterface|null $con a connection object
+     *
+     * @return \Propel\Runtime\DataFetcher\DataFetcherInterface
+     */
+    public static function execute(Criteria $criteria, ?ConnectionInterface $con = null): DataFetcherInterface
+    {
+        $executor = new CountQueryExecutor($criteria, $con);
+
+        return $executor->runCount();
+    }
+
+    /**
+     * Execute a count statement.
+     *
+     * @return \Propel\Runtime\DataFetcher\DataFetcherInterface
+     */
+    public function runCount(): DataFetcherInterface
```

Comment:
This public method is not used anywhere except in this class so it could be `protected`.

Reply:
Please check all children of `AbstractQueryExecutor`.

Reply:
This means that the query executors can only be used through their static `execute` method. So I have also changed all constructors to `protected`. Time will tell how this will fare in the long run.

Reply:
> This means that the query executors can only be used through their static execute method.
So you don't need another public method if you want to call executor only via static method.

> So I have also changed all constructors to protected.
Then we even don't have the possibility to get an instance of executor anywhere and we will not have the ability to call `runCount` even in the future.

Reply:
You just repeated what I said, right?

Reply:
Sorry, It looks like repetition now. But I understood your phrase "Time will tell how this will fare in the long run." that you want to keep it public for the future. And I wanted to underline that it should not be. Now I see that it is protected and my comment is excess. Sorry, for the misunderstanding.

Reply:
Ah, I see. Thank you for clearing it up.
Had to change the constructor from `AbstractQueryExecutor` (which most of the classes inherit directly) back to `public`, so that tests can override methods. I don't see a way around that.